### ISSUE TITLE
Specify host directory when using TART_EXECUTOR_HOST_DIR=true

### ISSUE THREAD

**Issue by rigellin73** [cirruslabs/gitlab-tart-executor#23]
Hello! I was exploring `TART_EXECUTOR_HOST_DIR=true` option for my pipelines and have some questions.

1. Is there any possibility to specify host dir instead of using tmp dir?
2. If there isn't, would you be interested in accepting a PR to support this?

We are trying to use tart and gitlab-tart-executor to optimise our pipelines. We have several big repos (up to 500 Gb) and usually we store them on volumes and don't download/delete sources every time we run pipeline. So for us it would be very useful to mount specific directory which already has sources.

**Comment by fkorotkov** [comment]
I guess it makes sense to have something like `TART_EXECUTOR_HOST_DIR_PATH` that will be used checked here:

https://github.com/cirruslabs/gitlab-tart-executor/blob/384811f0777a0a767fc71e18e9f3c3008a9d114f/internal/gitlab/env.go#L30-L32

Note that you'll need to add logic for clean up stage to not delete it:

https://github.com/cirruslabs/gitlab-tart-executor/blob/384811f0777a0a767fc71e18e9f3c3008a9d114f/internal/commands/cleanup/cleanup.go#L45

@edigaryev what do you think of `TART_EXECUTOR_HOST_DIR_PATH` in addition to `TART_EXECUTOR_HOST_DIR`?

**Comment by waddles** [comment]
IIUC you would need to have [`GIT_STRATEGY: fetch`](https://docs.gitlab.com/ee/ci/runners/configure_runners.html#git-strategy) set to gain any benefit from mounting the already cloned repo from the host. That would also imply you [can only have a concurrency of 1](https://docs.gitlab.com/ee/ci/runners/configure_runners.html#handling-concurrency) to prevent different jobs modifying the checkout during a build.

Since you can only run 1 job per host and the build dir is always checked out, I guess the only benefit of running in a VM is that you can have a consistent snapshot of tooling, otherwise I'd just run it on the host itself.

You already have the ability to pass a `--dir hostdir:/path/on/host` argument in the prepare stage that will mount the repo from the host into `/Volumes/My Shared Files/hostdir`, you just need to work out your directory structure so the runner finds your code where it expects it to be.

**Comment by rigellin73** [comment]
@edigaryev 
Yes, I want the ability to mount arbitrary directories. But it won't violate CI/CD principles, because:
- We set the builds_dir to be unique for each concurrent job using $CI_CONCURRENT_ID to ensure that two builds don’t share the same directory.
- Regarding clean slate, we use GIT_CLEAN_FLAGS=-ffdx to ensure the mounted repo copy is clean at the start of the build.
Both of these approaches are recommended by the official documentation here https://docs.gitlab.com/ee/ci/large_repositories/#git-clone-path 

Does it work like @waddles say in comment above?

> You already have the ability to pass a --dir hostdir:/path/on/host argument in the prepare stage that will mount the repo from the host into /Volumes/My Shared Files/hostdir, you just need to work out your directory structure so the runner finds your code where it expects it to be.

Basically, this is exactly what we need - an ability to pass --dir argument with any host path we want.

**Comment by edigaryev** [comment]
> Basically, this is exactly what we need - an ability to pass --dir argument with any host path we want.

Would https://github.com/cirruslabs/gitlab-tart-executor/pull/24 work for you?

**Comment by rigellin73** [comment]
> > Basically, this is exactly what we need - an ability to pass --dir argument with any host path we want.
> 
> Would #24 work for you?

Yes, thank you!
